src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -794,7 +797,23 @@ private string GetOutputPath(PackageBuilder builder, bool symbols = false, NuGet
 
         public static string GetInputFile(PackArgs packArgs)
         {
-            IEnumerable<string> files = packArgs.Arguments != null && packArgs.Arguments.Any() ? packArgs.Arguments : Directory.GetFiles(packArgs.CurrentDirectory);
+            bool usingDirectory = false;
+            if (packArgs.Arguments?.Count() == 1)
+            {
+                string first = packArgs.Arguments.First();
+                if (Directory.Exists(first))
+                {
+                    if (string.IsNullOrEmpty(packArgs.OutputDirectory))
+                    {
+                        packArgs.OutputDirectory = packArgs.CurrentDirectory;
```

Comment:
Seems like OutputDirectory should always return CurrentDirectory by default, unless and OutputDirectory actually gets set.
This seems like a weird place to update the OutputDirectory, inside of a GetInputFile function. What if another place needs the OutputDirectory updated too?